sklearn/utils/_array_api.py

```diff
@@ -357,6 +357,28 @@ def _expit(X):
     return 1.0 / (1.0 + xp.exp(-X))
 
 
+def _nanmin(X, axis=None):
+    xp, _ = get_namespace(X)
+    if _is_numpy_namespace(xp):
+        return numpy.nanmin(X, axis=axis)
```

Comment:
Unfortunately, this still needs `xp.asarray`

```suggestion
        return xp.asarray(numpy.nanmin(X, axis=axis))
```

At this point, `xp` can be a `numpy.array_api` Array which means the output needs to be one too.